generate.py

```diff
@@ -0,0 +1,593 @@
+        logger.error(f'Unknown User: {user_id}')
+        return
+
+
+def import_tarfile_in_course(tarfile_path, course_key, user_id):
+    """Helper method to import provided tarfile in the course."""
+
+    user = validate_user(user_id)
+    if not user:
+        return
+
+    data_root = path(settings.GITHUB_REPO_ROOT)
+    subdir = base64.urlsafe_b64encode(repr(course_key).encode('utf-8')).decode('utf-8')
+    course_dir = data_root / subdir
+
+    root_name = COURSE_ROOT
+    courselike_block = modulestore().get_course(course_key)
+    import_func = import_course_from_xml
+
+    # try-finally block for proper clean up after receiving file.
+    try:
+        tar_file = tarfile.open(tarfile_path)
+        try:
+            safetar_extractall(tar_file, (course_dir + '/'))
+        except SuspiciousOperation as exc:
+            logger.error(f'Unsafe tar file')
+            return
+        finally:
+            tar_file.close()
+        logger.info('Course tar file extracted. Verification step started')
+
+        dirpath = verify_root_name_exists(course_dir, root_name)
+        if not dirpath:
+            return
+
+        if not validate_course_olx(course_key, dirpath):
+            return
+
+        dirpath = os.path.relpath(dirpath, data_root)
+
+        logger.info(f'Extracted file verified. Updating course started')
+
+        courselike_items = import_func(
+            modulestore(), user.id,
+            settings.GITHUB_REPO_ROOT, [dirpath],
+            load_error_blocks=False,
+            static_content_store=contentstore(),
+            target_id=course_key,
+            verbose=True,
+        )
+
+        new_location = courselike_items[0].location
+        logger.debug('new course at %s', new_location)
+
+        logger.info(f'Course import successful')
+    except (CourseImportException, InvalidProctoringProvider, DuplicateCourseError) as known_exe:
+        logger.exception(f"Error while importing course: {known_exe}")
+    except Exception as exception:  # pylint: disable=broad-except
+        logger.exception(f"Error while importing course: {exception}")
+    finally:
+        if course_dir.isdir():
+            shutil.rmtree(course_dir)
+            logger.info('Temp data cleared')
+
+# -----------------------------------------------------------------------------
+
+
+User = get_user_model()
+
+USER_EMAIL = "edx@example.com"
+
+user = User.objects.get(email=USER_EMAIL)
+
+TARFILE_PATH = '/edx/src/taxonomy-sample-data/course.g4vmy6n2.tar.gz'
+
+SAMPLE_ORGS_COUNT = 2
+SAMPLE_ORG_NAME = "SampleTaxonomyOrg"
+COURSE_NAME = "Sample Taxonomy Course"
+COURSE_NUMBER = "STC1"
+COURSE_RUN = "2023_1"
+
+DISABLED_TAXONOMY_NAME = "DisabledTaxonomy"
+FLAT_TAXONOMY_NAME = "FlatTaxonomy"
+HIERARCHICAL_TAXONOMY_NAME = "HierarchicalTaxonomy"
+TWO_LEVEL_TAXONOMY_NAME = "TwoLevelTaxonomy"
+MULTI_ORG_TAXONOMY_NAME = "MultiOrgTaxonomy"
+
+IMPORT_OPEN_CANADA_TAXONOMY = False
+IMPORT_LIGHTCAST_SKILLS_TAXONOMY = False
```

Comment:
Can we just make these True by default? I think we'll generally want to have them.

Reply:
Sure thing